File: backend/shared/services/PermissionService.py
class PermissionService:
    """
    Minimal PermissionService implementation.
    Real logic should handle permission inheritance,
    departmentscoped access and dynamic evaluation.
    """

    def __init__(self):
        # placeholder for any initialization (e.g., DB connections)
        pass

    def get_user_permissions(self, user_id):
        """Return a list of permission identifiers for the given user."""
        return []

    def has_permission(self, user_id, permission):
        """Check if the user has a specific permission."""
        return permission in self.get_user_permissions(user_id)

File: src/backend/shared/services/PermissionService.js
export class PermissionService {
    /**
     * Returns the list of permission strings for a given user.
     * This placeholder implementation always returns an empty array.
     */
    async getPermissions(userId) {
        return [];
    }
    /**
     * Checks whether a user has a specific permission.
     * Placeholder always returns false.
     */
    async hasPermission(userId, permission) {
        return false;
    }
}
